Source/Core/VideoCommon/TextureDecoder_Common.cpp

```diff
@@ -320,7 +320,7 @@ static const char* texfmt[] = {
     "CZ16L", "0x3D", "0x3E", "0x3F",
 };
 
-static void TexDecoder_DrawOverlay(u8* dst, int width, int height, int texformat)
+static void TexDecoder_DrawOverlay(u8* __restrict dst, int width, int height, int texformat)
```

Comment:
Does this help? With only one pointer arg, there's nothing dst can alias to?

And in my little test bench w/gcc 7.1.0 x86_64 -O2 it produces the same code here with and without __restrict (https://godbolt.org/g/xoSfbL)